camera_image_pub_node:
  ros__parameters:
    img_resize_width: 640
    img_resize_height: 480
    cam_num: 0   # Put here a 0 if you only have one camera, put a 2 if you want your second cam.
    fps_pub: 15
    compression_format: "PNG" #Select one from: BMP, DIB, JPG, JPEG ,JPE, JP2, PNG, PBM, PGM, PPM, SR, RAS, TIFF or TIF.

image_color_filter_node:
  ros__parameters:
    img_resize_width: 640
    img_resize_height: 320 #these 2 params should match the camera_image_pub_node params.
    display_gui: True
    #filter_vals: [70, 30, 255, 220, 205, 75] # for the green of the simulator
    #filter_vals: [124, 94, 191, 118, 92, 59] # for the blue umbrella.
    filter_vals: [152, 119, 155, 69, 191, 68] # for the ECLA pink/magenta card.

image_operator_node:
  ros__parameters:
    debug_circle_min_radius: 3
    debug_circle_max_radius: 25
    debug_circle_scale_radius: 500.0  #70 for the simulation because greens were bigger and noiseless
    debug_circle_color: [255, 0, 255]   # GBR

qr_code_detector_node:
  ros__parameters:
    side_threshold: 5.0
    display_gui: True

sink_motors_node:
  ros__parameters:
    only_rotation_mode: False
    only_display_mode: False
    goal_x_dist: 0.0    #Between the range [-1.0, 1.0] (pixels)
    goal_pix_diag: 15.0   #Between the range [-1.0, 1.0] (pixels)
    goal_x_threshold: 0.12         #Between the range [-1.0, 1.0]
    goal_pix_diag_threshold: 0.05   #Between the range [-1.0, 1.0]
    kp_w: 0.35   #Proportional controller constants for v and w
    kp_v: 0.02   #kp_v is proportional to the object size.
    target_lost_timeout: 3000 #ms for the robot to start searching the target when is not seen.


